### DISCUSSION TITLE
Using metadata values to control the visibility of slides

### DISCUSSION THREAD

**Discussion by ellafm** [quarto-dev/quarto-cli#10487] [Q&A] [answered]
### Description

I want to produce lecture slides in two versions. One version contains answers to the problems, and the other does not. The code shown below achieves this, with the metadata value `answer_slides` controlling whether the solutions appear in the rendered presentation.

But putting a div around a whole slide looks like a hack to me. Is it the recommended approach, or are there better ways to achieve the same effect. For example, could it be done by defining a `.answer` class and using a lua filter?

````qmd
---
title: "Some title"
author: "An author"
jupyter: python3
format: 
    revealjs:
        theme: default
answer_slides: false
---

## First slide

Some content

::: {.content-hidden unless-meta="answer_slides"}
## Second slide

This slide will be hidden unless `answer_slides` is set to `true`
:::
````

**Comment by mcanouil** [comment, marked as answer]
That's how it is documented and how this feature works: https://quarto.org/docs/authoring/conditional.html

Note that in this case I would use profile. https://quarto.org/docs/projects/profiles.html

````qmd
---
title: "Some title"
author: "An author"
engine: markdown
format: 
  revealjs:
    theme: default
---

## First slide

Some content

::: {.content-visible when-profile="answer"}
## Second slide

This slide will be hidden unless `answer_slides` is set to `true`
:::
````

```sh
quarto render index.qmd
quarto render index.qmd --profile answer --output with-answers.html
```

**Comment by finkelshtein** [comment]
@mcanouil Would it possible to get both files (without and with answers) on the same website?
I currently publish each Lab of a course without answers, and in a couple of days I publish it with answers. 
Both versions of each Lab should be present on the website (so that students would be able to access the versions without answers later on).

**Reply by mcanouil** [reply]
You could use `_metadata.yml` and symbolic links such as:

```txt
├── questions
│   ├── _metadata.yml
│   └── subject1.qmd
├── questions.qmd
├── answers
│   ├── _metadata.yml
│   └── subject1.qmd -> ../questions/subject1.qmd
├── answers.qmd
├── _quarto.yml
└── index.qmd
```

**Reply by mcanouil** [reply]
Here is the demo:

- https://github.com/quarto-dev/quarto-cli/discussions/11635
